backend/static/js/search-results.js

```diff
@@ -37,59 +36,88 @@ function attachEventHandlersPagination() {
-  We cannot just loop over every form element, because buttons and hidden inputs should be handled differently.
-*/
+/**
+ * Attach an event listener to form 'reset' event.
+ * On form reset, wipe all fields and return them to their default states.
+ */
 function attachEventHandlersReset() {
+  // The standard form behavior does _not_ reset to fully empty fields if the user had just made a search.
   FORM.addEventListener('reset', (e) => {
     e.preventDefault();
+
+    resetCheckboxes();
+    resetCogOver();
+
     // Empty out textareas
     var text_areas = document.getElementsByTagName('textarea');
     Array.from(text_areas).forEach((input) => {
       input.value = '';
     });
-    // Uncheck checkboxes
-    var checkboxes = document.querySelectorAll(
-      '[type="checkbox"], [type="radio"]'
+    // Wipe date fields
+    var date_inputs = document.querySelectorAll(
+      'div.usa-date-picker__wrapper > input'
     );
-    Array.from(checkboxes).forEach((checkbox) => {
-      checkbox.checked = false;
+    Array.from(date_inputs).forEach((date_input) => {
+      date_input.value = '';
     });
-    // Wipe FAC release dates
-    var start_date = document.getElementById('start-date');
-    var end_date = document.getElementById('end-date');
-    start_date.value = '';
-    end_date.value = '';
-    // Reset Cog/Over dropdown
-    var default_cog_over_option = document.getElementById(
-      'cog_or_oversight--both'
-    );
-    default_cog_over_option.selected = true;
-    // Wipe agency name
-    var agency_name = document.getElementById('agency-name');
-    agency_name.value = '';
     // Reset state dropdown
     var default_state_option = document.getElementById('state--none');
     default_state_option.selected = true;
+    // Reset fiscal year end month
+    var default_FY_option = document.getElementById('fy-month--none');
+    default_FY_option.selected = true;
 
     FORM.reset();
   });
 }
 
-/*
-  Get the table headers and their sort buttons.
-  Using the ID of the table headers, attach event handlers to re-submit the form when clicking them.
-  This will reload the page with the associated sort values, so that searches appear to sort across many pages.
-
-  If the sort buttons get clicked, we have one of three states:
-    1. We are not sorting, and we move to sort in descending.
-    2. We are sorting by descending, and we move to sort in ascending.
-    3. We are sorting by ascending, and we move to no sort at all.
-      i. In this case, we wipe both the order_by and order_direction fields. Or, it will maintain the values and sort anyway.
-  In any case, we want to reset the page number to one.
-*/
+/**
+ * Reset the checkboxes. Clears audit year, entity type, etc.
+ * Re-check audit year 2023.
+ */
+function resetCheckboxes() {
+  var checkboxes = document.querySelectorAll(
+    '[type="checkbox"], [type="radio"]'
+  );
+  var audit_year_2023 = document.getElementById('audit-year-2023');
```

Comment:
nitpick: is it possible to refactor this so that it targets the second checkbox (skipping "all years") rather than explicitly targeting 2023? Might save our future selves an extra step/headache when rolling over to a new submission year

Reply:
For sure! I thought about doing this, and promptly forgot. Good call!

I looked at maybe sorting the list of audit years to grab the most recent audit year (the largest number), but that seems somewhat excessive here. So, I just recheck the second box.